cms/static/js/views/utils/view_utils.js

```diff
@@ -173,6 +174,33 @@ define(["jquery", "underscore", "gettext", "js/views/feedback_notification", "js
             return false;
         };
 
+        keywordValidator = (function () {
+            var regexp = /%%+[^%]+%%/g;
+            var keywords = ['%%USER_ID%%', '%%USER_FULLNAME%%', '%%COURSE_DISPLAY_NAME%%', '%%COURSE_END_DATE%%'];
+            var validate = function (string) {
+                var regex_match = string.match(regexp);
+                var found_keywords = regex_match == null ? [] : regex_match;
+                var invalid_keywords = [];
+                var num_found = found_keywords.length;
+                var curr_keyword;
+
+                $.each(found_keywords, function(i, keyword) {
```

Comment:
It feels like `$.map` would be a more natural fit here [1].

```javascript
var keywordsInvalid = $.map(keywordsFound, function (keyword) {
    if ($.inArray(keyword, keywordsSupported) === -1) {
        return keyword;
    } else {
        // return `null` or `undefined` to remove an element
        return undefined;
    }
});
```

[1] http://api.jquery.com/jquery.map/